curriculum/challenges/english/25-front-end-development/lab-book-organizer/67172b43f84bcd2dec238a3d.md

```diff
@@ -0,0 +1,169 @@
+
+4. The `sortByYear` function should return `-1` if the `releaseYear` of the first book is smaller than that of the second book.
+
+5. The `sortByYear` function should return `1` if the `releaseYear` of the first book is bigger than that of the second book.
+
+6. The `sortByYear` function should return `0` if both `releaseYear` values are equal.
+
+7. You should filter out books written after a certain year such as 1950 from the `books` array and save the filtered array in a new array named `filteredBooks`.
+
+8. You should sort the books in the `filteredBooks` array according to their `releaseYear` in ascending order and save the sorted array in a new array named `filteredBooksSorted`.
+
+# --hints--
+
+You should have a function `sortByYear` in your code.
+
+```js
+assert.isFunction(sortByYear);
+```
+
+Your `sortByYear` function should take two parameters.
+
+```js
+assert.lengthOf(sortByYear, 2);
+```
+
+Your `sortByYear` function should return `-1` if the `releaseYear` of `book1` object is smaller than that of the `book2` object, `1` if the `releaseYear` of `book1` object is larger than that of the `book2` object, and `0` in all other scenarios.
+
+```js
+assert.equal(sortByYear({releaseYear: 1913}, {releaseYear: 1925}), -1);
+assert.equal(sortByYear({releaseYear: 1925}, {releaseYear: 1913}), 1);
+assert.equal(sortByYear({releaseYear: 1925}, {releaseYear: 1925}), 0);
+```
+
+You should have an array `books` in your code.
+
+```js
+assert.isArray(books)
+```
+
+Your `books` array should have at least three book objects.
+
+```js
+assert.isAtLeast(books.length, 3)
+```
+
+Your `books` array should contain objects each with a string `title`, another string `authorName`, and a number `releaseYear`.
+
+```js
+
+books.forEach((book) => {
+  assert.isObject(book);
+  assert.sameMembers(Object.keys(book), ["title", "authorName", "releaseYear"]);
+  assert.isString(book.title);
+  assert.isNotEmpty(book.title);
+  assert.isString(book.authorName);
+  assert.isNotEmpty(book.authorName);
+  assert.isNumber(book.releaseYear);
+})
+```
+
+You should have an array `filteredBooks` in your code.
+
+```js
+assert.isArray(filteredBooks)
+```
+
+The `filteredBooks` array should have some of the books filtered out from the `books` array and not be empty.
+
+```js
+assert.isBelow(filteredBooks.length, books.length)
+assert.isNotEmpty(filteredBooks)
+```
+
+You should have an array `filteredBooksSorted` in your code.
+
+```js
+assert.isArray(filteredBooksSorted)
+```
+
+The `filteredBooksSorted` array should have the same length of `filteredBooks`.
+
+```js
+assert.lengthOf(filteredBooksSorted, filteredBooks.length)
+```
+
+You should call the `sort` higher order function by passing the `sortByYear` callback function on the `filteredBooks` array.
+
+```js
+assert.match(__helpers.removeJSComments(code), //(const|let)\s+filteredBooksSorted\s*=\s*filteredBooks\s*\.\s*sort\s*\(\s*sortByYear\s*\)//)
+```
```

Comment:
this doesn't looks right to me. 

It should probably be this, but you should test it on your end to make sure

````suggestion
```js
assert.match(__helpers.removeJSComments(code), /(const|let)\s+filteredBooksSorted\s*=\s*filteredBooks\s*\.\s*sort\s*\(\s*sortByYear\s*\)/);
```
````

Reply:
My bad, I tested the regex and then copied and pasted it to make the suggestion and I didn't realize that I left the forward slashes there.